### ISSUE TITLE
`indentation_linter()` doesn't allow indention inside functions

### ISSUE THREAD

**Issue by simensol** [r-lib/lintr#1824]
I configured lintr using the example in the documentation:

```R
lintr::use_lintr(type = "tidyverse")
```

Executing `lintr::lint_package()` with the following code:

```R
load_stops <- function(file_path) {
  data <- xml2::read_xml(file_path)

  xml2::xml_ns_strip(data)

  stop_places <- xml2::xml_find_all(data, "//StopPlace[TransportMode='bus']")

  tibble::tibble(
    id = stop_places %>% xml2::xml_attr("id"),
    name = stop_places %>%
      xml2::xml_find_first("./Name") %>%
      xml2::xml_text()
  )
}
```

makes lintr report:

<img width="288" alt="RStudio" src="https://user-images.githubusercontent.com/4491653/206682298-a5e0de05-0854-411e-aba3-481c3e56b081.png">

The only way to make lintr happy, is to remove all indention:

```R
load_stops <- function(file_path) {
data <- xml2::read_xml(file_path)

xml2::xml_ns_strip(data)

stop_places <- xml2::xml_find_all(data, "//StopPlace[TransportMode='bus']")

tibble::tibble(
id = stop_places %>% xml2::xml_attr("id"),
name = stop_places %>%
xml2::xml_find_first("./Name") %>%
xml2::xml_text()
)
}
```

What happens here?

EDIT:

When I lint the same code using the GitHub workflow created by `usethis::use_github_action("lint")`, all errors are gone. Linting locally using `lintr::lint_package()` produces the errors mentioned above.

**Comment by simensol** [comment]
When I lint the same code using the GitHub workflow created by `usethis::use_github_action("lint")`, all errors are gone. Linting locally using `lintr::lint_package()` produces the errors mentioned above.

**Comment by IndrajeetPatil** [comment]
I can't reproduce this behaviour with the current `main`-branch:

``` r
library(lintr)

'load_stops <- function(file_path) {
  data <- xml2::read_xml(file_path)
  
  xml2::xml_ns_strip(data)
  
  stop_places <- xml2::xml_find_all(data, "//StopPlace[TransportMode=\'bus\']")
  
  tibble::tibble(
    id = stop_places %>% xml2::xml_attr("id"),
    name = stop_places %>%
      xml2::xml_find_first("./Name") %>%
      xml2::xml_text()
  )
}' -> code

lint(
  text = code,
  linters = indentation_linter()
)
```

<sup>Created on 2022-12-09 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

<details style="margin-bottom:10px;">
<summary>
Session info
</summary>

``` r
sessioninfo::session_info()
#> ─ Session info ───────────────────────────────────────────────────────────────
#>  setting  value
#>  version  R version 4.2.2 (2022-10-31)
#>  os       macOS Ventura 13.0.1
#>  system   aarch64, darwin20
#>  ui       X11
#>  language (EN)
#>  collate  en_US.UTF-8
#>  ctype    en_US.UTF-8
#>  tz       Europe/Berlin
#>  date     2022-12-09
#>  pandoc   2.19.2 @ /Applications/RStudio.app/Contents/MacOS/quarto/bin/tools/ (via rmarkdown)
#> 
#> ─ Packages ───────────────────────────────────────────────────────────────────
#>  package      * version       date (UTC) lib source
#>  backports      1.4.1         2021-12-13 [1] CRAN (R 4.2.0)
#>  callr          3.7.3         2022-11-02 [1] CRAN (R 4.2.0)
#>  cli            3.4.1         2022-09-23 [1] CRAN (R 4.2.0)
#>  crayon         1.5.2         2022-09-29 [1] CRAN (R 4.2.1)
#>  cyclocomp      1.1.0         2016-09-10 [1] CRAN (R 4.2.0)
#>  desc           1.4.2         2022-09-08 [1] CRAN (R 4.2.1)
#>  digest         0.6.30        2022-10-18 [1] CRAN (R 4.2.1)
#>  evaluate       0.18          2022-11-07 [1] CRAN (R 4.2.2)
#>  fastmap        1.1.0         2021-01-25 [1] CRAN (R 4.2.0)
#>  fs             1.5.2         2021-12-08 [1] CRAN (R 4.2.0)
#>  glue           1.6.2         2022-02-24 [1] CRAN (R 4.2.0)
#>  highr          0.9           2021-04-16 [1] CRAN (R 4.2.0)
#>  htmltools      0.5.4         2022-12-07 [1] CRAN (R 4.2.2)
#>  knitr          1.41          2022-11-18 [1] CRAN (R 4.2.2)
#>  lazyeval       0.2.2         2019-03-15 [1] CRAN (R 4.2.0)
#>  lifecycle      1.0.3         2022-10-07 [1] CRAN (R 4.2.1)
#>  lintr        * 3.0.2.9000    2022-12-09 [1] local
#>  magrittr       2.0.3         2022-03-30 [1] CRAN (R 4.2.0)
#>  processx       3.8.0         2022-10-26 [1] CRAN (R 4.2.1)
#>  ps             1.7.2         2022-10-26 [1] CRAN (R 4.2.0)
#>  purrr          0.9000.0.9000 2022-11-30 [1] Github (tidyverse/purrr@55c9a8a)
#>  R.cache        0.16.0        2022-07-21 [1] CRAN (R 4.2.0)
#>  R.methodsS3    1.8.2         2022-06-13 [1] CRAN (R 4.2.0)
#>  R.oo           1.25.0        2022-06-12 [1] CRAN (R 4.2.0)
#>  R.utils        2.12.2        2022-11-11 [1] CRAN (R 4.2.2)
#>  R6             2.5.1.9000    2022-10-27 [1] local
#>  remotes        2.4.2         2021-11-30 [1] CRAN (R 4.2.0)
#>  reprex         2.0.2         2022-08-17 [1] CRAN (R 4.2.1)
#>  rex            1.2.1         2021-11-26 [1] CRAN (R 4.2.0)
#>  rlang          1.0.6         2022-09-24 [1] CRAN (R 4.2.1)
#>  rmarkdown      2.18          2022-11-09 [1] CRAN (R 4.2.2)
#>  rprojroot      2.0.3         2022-04-02 [1] CRAN (R 4.2.0)
#>  rstudioapi     0.14          2022-08-22 [1] CRAN (R 4.2.1)
#>  sessioninfo    1.2.2         2021-12-06 [1] CRAN (R 4.2.0)
#>  stringi        1.7.8         2022-07-11 [1] CRAN (R 4.2.1)
#>  stringr        1.5.0         2022-12-02 [1] CRAN (R 4.2.2)
#>  styler         1.8.1.9000    2022-11-26 [1] Github (r-lib/styler@44d4787)
#>  vctrs          0.5.1         2022-11-16 [1] CRAN (R 4.2.2)
#>  withr          2.5.0         2022-03-03 [1] CRAN (R 4.2.0)
#>  xfun           0.35          2022-11-16 [1] CRAN (R 4.2.2)
#>  xml2           1.3.3.9000    2022-10-10 [1] local
#>  xmlparsedata   1.0.5         2021-03-06 [1] CRAN (R 4.2.0)
#>  yaml           2.3.6         2022-10-18 [1] CRAN (R 4.2.1)
#> 
#>  [1] /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/library
#> 
#> ──────────────────────────────────────────────────────────────────────────────
```

</details>

**Comment by simensol** [comment]
Reinstalling lintr solved the issue.
